integration-libs/cdc/schematics/add-cdc/index_spec.ts

```diff
@@ -121,19 +112,115 @@ describe('Spartacus CDC schematics: ng-add', () => {
         appTree
       );
     });
-
     it('should not create any of the feature modules', () => {
       expect(appTree.exists(cdcFeatureModulePath)).toBeFalsy();
     });
   });
-
-  describe('CDC feature', () => {
+  describe('CDC-B2C feature', () => {
     describe('validation of jsSDKUrl', () => {
       beforeEach(async () => {
         appTree = await schematicRunner.runSchematic(
           'ng-add',
           {
             ...cdcFeatureOptions,
+            features: [USER_ACCOUNT_FEATURE_NAME, USER_PROFILE_FEATURE_NAME],
+          },
+          appTree
+        );
+        appTree = await schematicRunner.runSchematic(
+          'ng-add',
+          { ...cdcFeatureOptions, javascriptUrl: '<dc>.gigya.com/<api-key>' },
+          appTree
+        );
+      });
+      it('should set the given javascriptUrl', async () => {
+        const featureModule = appTree.readContent(cdcFeatureModulePath);
+        expect(featureModule).toMatchSnapshot();
+      });
+    });
+    describe('general setup', () => {
+      beforeEach(async () => {
+        appTree = await schematicRunner.runSchematic(
+          'ng-add',
+          {
+            ...cdcFeatureOptions,
+            features: [USER_ACCOUNT_FEATURE_NAME, USER_PROFILE_FEATURE_NAME],
+          },
+          appTree
+        );
+        appTree = await schematicRunner.runSchematic(
+          'ng-add',
+          cdcFeatureOptions,
+          appTree
+        );
+      });
+      it('should install necessary Spartacus libraries', () => {
+        const packageJson = JSON.parse(appTree.readContent('package.json'));
+        let dependencies: Record<string, string> = {};
+        dependencies = { ...packageJson.dependencies };
+        dependencies = { ...dependencies, ...packageJson.devDependencies };
+        for (const toAdd in peerDependencies) {
+          // skip the SPARTACUS_SCHEMATICS, as those are added only when running by the Angular CLI, and not in the testing environment
+          if (
+            !peerDependencies.hasOwnProperty(toAdd) ||
+            toAdd === SPARTACUS_SCHEMATICS
+          ) {
+            continue;
+          }
+          // TODO: after 4.0: use this test, as we'll have synced versions between lib's and root package.json
+          // const expectedVersion = (peerDependencies as Record<
+          //   string,
+          //   string
+          // >)[toAdd];
+          const expectedDependency = dependencies[toAdd];
+          expect(expectedDependency).toBeTruthy();
+          // expect(expectedDependency).toEqual(expectedVersion);
```

Comment:
[NITPICK]
I'm aware your bug fix is urgent and I don't want to block this PR because of this copy-pasted (outdated) TODO.

But do you think that afterwards (when it's merged) you could take a look at it and try to fulfill the recommendation of this TODO and the remove the TODO comment (in a separate PR), so we clean up the code but not pollute it with TODOs even more? I hope it's doable, as the comment says "after 4.0" and we're now "6.x" 😉 

Btw. SonarQube in CI likely won't let you merge this PR, becasue it contains //TODO comment.
I suggest to create a ticket for it and put `// CXSPA-xxxx:`  (with the number of the ticket you create) instead of `//TODO: ` to workaround the SonarQube.

Reply:
done - have created [CXSPA-4872](https://jira.tools.sap/browse/CXSPA-4872)